iridauploader/core/api_handler.py

```diff
@@ -161,18 +165,22 @@ def upload_sequencing_run(sequencing_run, directory_status, upload_mode):
         for project in sequencing_run.project_list:
             # loop through samples
             for sample in project.sample_list:
-                logging.info("Uploading to Sample {} on Project {}".format(sample.sample_name, project.id))
-                # upload files
-                api_instance.send_sequence_files(sequence_file=sample.sequence_file,
-                                                 sample_name=sample.sample_name,
-                                                 project_id=project.id,
-                                                 upload_id=run_id,
-                                                 upload_mode=upload_mode)
-                # Update status file on progress
-                directory_status.set_sample_uploaded(sample_name=sample.sample_name,
+                if sample.skip:
+                    logging.info("Skipping Sample {} on Project {}, already uploaded."
+                                 "".format(sample.sample_name, project.id))
```

Comment:
If we're skipping a sample, it needs to have its sample status set to uploaded.  Otherwise in the status file skipped files are listing `"Uploaded": "False`.  If a run gets cancelled 2x, the skipped files will get uploaded again.

Reply:
Fixed. Also simplifies the loop a bit.